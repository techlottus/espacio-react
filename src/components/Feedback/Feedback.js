import React, { useEffect } from 'react';
import "./Feedback.scss";

export const Feedback = ({data, onRight, text, html=() => {return}, isNoti=false}) => {

    const feedbackRef = React.createRef();
  
    useEffect(() => {
      feedbackRef.current.data = {
        left: data.left || {},
        right: data.right || {},
        type: data.type || '',
        isTextEvent: data.isTextEvent,
        textEvent: data.textEvent || '',
      };
      
    },[data,text]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
      feedbackRef.current.addEventListener('onRight', onRight)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
    
    return (
      <>
        <lottus-feedback ref={feedbackRef}> 
          <div content="" className={`feedback-content ${isNoti ? "feedbackcontentmessagetext" : ""}`}>{text || html()}</div>
        </lottus-feedback>
      </>
    )
  }
