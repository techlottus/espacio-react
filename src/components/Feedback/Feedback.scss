@import "../../App.scss";

.feedback-content {
  font-family: "Nunito", sans-serif;
  line-height: pixeltorem(19px);
  font-size: pixeltorem(14px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    font-size: pixeltorem(16px);
    line-height: pixeltorem(22px);
  }
}

.feedbackcontentmessagetext {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}