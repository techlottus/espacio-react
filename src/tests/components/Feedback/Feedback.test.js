import React from "react";
import { mount } from "enzyme";
import { Feedback } from "../../../components/Feedback/Feedback";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    left: null,
    right: null,
    type: null,
    isTextEvent: null,
    textEvent: null,
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Feedback data={initState.data} />
    </Provider>
  );
};

describe("Test <Feedback />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
