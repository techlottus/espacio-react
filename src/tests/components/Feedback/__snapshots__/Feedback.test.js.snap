// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Feedback /> should init component 1`] = `
<lottus-feedback>
  <div
    className="feedback-content "
    content=""
  />
</lottus-feedback>
`;
